### PR TITLE
Bugfix for TPMAttestationValidator validateSubjectAlternativeName NPE

### PR DESCRIPTION
**Pull request by ynojima**
#80 by Kris plus some refactoring

### CODE REVIEW

File: webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/tpm/TPMAttestationStatementValidator.java [hunk 1]
```diff
@@ -293,57 +270,56 @@ private void validateSubjectAlternativeName(X509Certificate certificate) throws
...
 
-                //result.put(aid, value);
-                result.add(new KeyValue(aid, value));
-            }
-        } catch (Exception e) {
-            //
+    private String decodeAttr(byte[] attr) throws IOException {
+        String value = null;
+        if(attr != null){
+            Asn1Utf8String partNumberUtf8String = new Asn1Utf8String();
```
Comment (kvandermast) [inline webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/tpm/TPMAttestationStatementValidator.java]: I would rename `partNumberUtf8String`to `attrAsn1Utf8String`

File: webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/tpm/TPMAttestationStatementValidator.java [hunk 2]
```diff
@@ -293,57 +270,56 @@ private void validateSubjectAlternativeName(X509Certificate certificate) throws
...
-            return key;
+                result.put(aid, value);
+            }
+            return result;
         }
-
-        public void setKey(String key) {
-            this.key = key;
+        catch (NamingException e){
+            throw new BadAttestationStatementException("Failed to parse attestation certificate", e);
```
Comment (kvandermast) [inline webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/tpm/TPMAttestationStatementValidator.java]: Change the exception message to "Failed to parse the TPM attestation attributes"?

Comment (ynojima) [reply webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/tpm/TPMAttestationStatementValidator.java]: Thank you for suggestion. will correct.

**Comment by ynojima** [comment]
Thank you for your review. I've pushed the fix.
I'll merge this into master when the Travis-CI is fixed.

**Pull request merged**